rust/connlib/clients/android/src/lib.rs

```diff
@@ -279,29 +247,9 @@ impl Callbacks for CallbackHandler {
         })
     }
 
-    fn on_error(&self, error: &Error) -> Result<(), Self::Error> {
-        self.env(|mut env| {
-            let error = env.new_string(error.to_string()).map_err(|source| {
-                CallbackError::NewStringFailed {
-                    name: "error",
-                    source,
-                }
-            })?;
-            call_method(
-                &mut env,
-                &self.callback_handler,
-                "onError",
-                "(Ljava/lang/String;)Z",
-                &[JValue::from(&error)],
-            )
-        })
-    }
-
     fn roll_log_file(&self) -> Option<PathBuf> {
         self.handle.roll_to_new_file().unwrap_or_else(|e| {
-            tracing::debug!("Failed to roll over to new file: {e}");
-            let _ = self.on_error(&Error::LogFileRollError(e));
```

Comment:
Literally everywhere we issue the `on_error` callback, we also log an error right before it. Good riddance!

Reply:
I guess it makes sense if there won't be any errors from connlib that a user could possibly fix. "Check your Wi-Fi / Ethernet" is already signaled by on_disconnect, right?

Reply:
Could we do this in a separate PR perhaps? I didn't look at the changes yet but if possible, splitting refactorings off from features allows for a cleaner history.

Reply:
@thomaseizinger Yeah, that's probably a good idea

Reply:
@ReactorScram That's correct, `onError` doesn't trigger any user-actionable errors.

Reply:
@thomaseizinger Updated. This PR only contains the DNS changes.